votepy/algorithms/base_ilp.py

```diff
@@ -0,0 +1,32 @@
+from votepy.ordinal_election import OrdinalElection
+from votepy.algorithms.base_algorithm import BaseAlgorithm
+from votepy.structure.structure import algo
+from votepy.generic_ilp import CPLEX, Gurobi
+
+
+@algo(name='ilp')
+class ILP(BaseAlgorithm):
+
+    def __init__(self, solver=Gurobi):
+        """Base class for ILP algorithms"""
+        super().__init__()
+        self.solver = solver
+
+    def prepare(self, define_model):
+        self.define_model = define_model
+        super().prepare()
```

Comment:
You can add description/example/type hint of the `define_model`. At the first glance I thought it is a string descripton of the ILP.

Reply:
Yeah, I forgot about that, sorry